src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -825,7 +825,7 @@ void handleConnack(MqttConnAckMessage msg) {
           this.connectHandler.handle(Future.succeededFuture(msg));
         } else {
           MqttConnectionException exception = new MqttConnectionException(msg.code());
-          log.error("Connection refused by the server");
+          log.warn(String.format("Connection refused by the server - code: %s", msg.code()));
```

Comment:
imho it's an error because at MQTT application level your connection to the server didn't go well. It's something that the application should take care of because it's not connected to the server successfully. The warning in this case sounds to me something like "pay attention something went wrong but you can continue without any action". I would stick with error log level but updating the message as you are proposing with MQTT return code.

Reply:
Well the issue with that just is, your log might get spammed with error messages, although your application might be able to take care of this. So you need to manually suppress error messages for this MqttClientImpl.

But I did amend the PR and changed it back to "error". The error code is more important to me 😉